daemon/daemon.go

```diff
@@ -990,7 +990,9 @@ func NewDaemon(ctx context.Context, config *config.Config, pluginStore *plugin.S
 		// It is painful. Add WithBlock can prevent the edge case. And
 		// n common case, the containerd will be serving in shortly.
 		// It is not harm to add WithBlock for containerd connection.
-		grpc.WithBlock(),
+		//
+		// TODO(thaJeztah): update this list once https://github.com/containerd/containerd/pull/10250/commits/63b46881753588624b2eac986660458318581330 is in the 1.7 release.
+		grpc.WithBlock(), //nolint:staticcheck // Ignore SA1019: grpc.WithBlock is deprecated: this DialOption is not supported by NewClient. Will be supported throughout 1.x.
```

Comment:
I went for suppressing the deprecation for now, because it's supported throughout 1.x, and we probably want to switch over once containerd 1.7 client has the linked patches.

I also opened a separate PR to align our options more with containerd's defaults, because it looks like we diverged;

- https://github.com/moby/moby/pull/48617